packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -1025,7 +1025,8 @@ def check_majority_possible(
         if len(votes_by_participant) == 0:
             return
 
-        vote_count = Counter(votes_by_participant.values())
+        votes = votes_by_participant.values()
+        vote_count = Counter(tuple(sorted(v.data.items())) for v in votes)
```

Comment:
we're still depending on the fact that the `.data` attribute is correctly implemented by the developer, as the base class `BaseTxPayload` always returns an empty dict.  I am not a fan of this empty dict, but remember there was a reason for this - please inform me of this reason so we can reconsider.

Reply:
Yes there are payloads that may not have any data at some point. I don't see how that warrants returning an empty dict from `BaseTxPayload`, if such is the case the developer can implement it themselves. Would it not be better to make the `data` an abstractmethod to force the developer to implement it on concrete classes, instead of allowing for the possibility to forget to implement it, hence return an empty dict and potentially run into unexpected behaviour much further downstream? 

Also I don't quite follow the final remark. If we use `.keys()`  then the values could still be different and we would equate such payloads equal?

Reply:
Some payloads return empty dicts on purpose. I agree with your suggestion on enforcing the implementation of `data`, but this is a separate issue that does not concern this PR. However, for this PR to be complete, I believe that we should also account for empty dicts in this method.

Re the `.keys()` I have misunderstood.

Reply:
ok you asked in PM and was settled bu works the same for empty dict